### ISSUE TITLE
Allow User To Schedule How Long They Need a Service For

### ISSUE THREAD

**Issue by freddyyeddy** [alextselegidis/easyappointments#720]
My use case is for interpreting services and cannot be a fixed amount of time per service scheduling. So I was wondering if there was a way for a person to choose how long they needed a service for. Instead of only allowing a fixed amount of time per service.

**Comment by yrammos** [comment]
This would be wonderful but would involve a fundamental redesign because, currently, available dates and times (step #2 of the booking process) are calculated on the basis of the selected service (step #1) and its duration.

**Comment by syn6UK** [comment]
Yes but theoretically, could it be done with some redesign? Can you advise on where in the code the calculation takes place so I can take a stab at this?

I'm trying to build an API for a number of Apps that utilise variable appointment times, and acuity or calendly don't even offer this.

**Comment by alextselegidis** [comment]
Hello! 

Yes this could be done with some customizations. 

The availability generation code resides in the Availability.php library. 

<small>
<table>
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator <br>
&nbsp;&nbsp;<a href="https://alextselegidis.com">Need a customization? Contact me in person!</a></td>
</tr>
</table>
</small>

**Comment by syn6UK** [comment]
No sorry, this is for a personal project, I'm tied up on commercial work at the moment.
